tests/test_ext_autodoc_autofunction.py

```diff
@@ -79,10 +79,15 @@ def test_method(app):
 @pytest.mark.sphinx('html', testroot='ext-autodoc')
 def test_builtin_function(app):
     actual = do_autodoc(app, 'function', 'os.umask')
-    assert list(actual) == [
+    actual = list(actual)
+    assert actual[:3] == [
         '',
         '.. py:function:: umask(mask, /)',
         '   :module: os',
+    ]
+    assert actual[3].startswith("   :canonical: ")
+    assert actual[3].endswith(".umask")
+    assert actual[-3:] == [
         '',
```

Comment:
How about this? (of course, importing `os.umask` is needed)
```suggestion
    assert list(actual) == [
        '',
        '.. py:function:: umask(mask, /)',
        '   :module: os',
        '   :canonical: %s.umask' % os.umask.__module__,
        '',
```

Reply:
Clean and flexible, I like it!